app/src/main/java/org/sopt/at/SignUpActivity.kt

```diff
@@ -0,0 +1,334 @@
+                            val intent = Intent().apply {
+                                putExtra("id", id)
+                                putExtra("password", password)
+                            }
+                            setResult(RESULT_OK, intent)
+                            finish()
+                        } else {
+                            isError = true
+                            scope.launch {
+                                snackbarHostState.showSnackbar("비밀번호는 영문, 숫자, 특수문자(~!@#\$&^&*) 조합 8~15자리입니다.")
+                            }
+                        }
+                    }
+                    Column(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .height(48.dp)
+                    ) {
+                        IconButton(
+                            onClick = {
+                                if (step == "비밀번호") {
+                                    isError = false
+                                    step = "아이디"
+                                } else {
+                                    finish()
+                                }
+                            },
+                            modifier = Modifier.size(48.dp)
+                        ) {
+                            Icon(
+                                imageVector = Icons.AutoMirrored.Default.ArrowBack,
+                                contentDescription = "뒤로가기",
+                                tint = Color.White
+                            )
+                        }
+                    }
+                    SignUpView(
+                        modifier = Modifier.padding(innerPadding),
+                        onNext = onNext,
+                        signUp = signUp,
+                        step = step,
+                        isError = isError
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun SignUpView(
+    modifier: Modifier,
+    onNext: (String) -> Unit,
+    signUp: (String, String) -> Unit,
+    step: String,
+    isError: Boolean
+) {
+    var id by remember { mutableStateOf("") }
+    var password by remember { mutableStateOf("") }
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .background(color = Color.Black),
+        verticalArrangement = Arrangement.Top,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+
+        if (step == "아이디") {
+            IdInputView(
+                text = id,
+                onValueChange = { id = it },
+                onNext = { onNext(id) },
+                isError = isError
+            )
+        } else if (step == "비밀번호") {
+            PasswordInputView(
+                text = password,
+                onValueChange = { password = it },
+                onSignUp = { signUp(id, password) },
+                isError = isError
+            )
+        }
```

Comment:
```suggestion
when (step) {
    SignUpStep.ID -> IdInputView(...)
    SignUpStep.PASSWORD -> PasswordInputView(...)
}
```
분기 처리를 할 때에는 if,else보다는 enum과 함께 when을 사용하는 것이 가독성이 더 높아집니다 !